corehq/apps/app_manager/fields.py

```diff
@@ -624,75 +620,30 @@ def get_form_rmi_response(self):
             response = response._asdict()
         return response
 
-    def _get_cases_for_apps(self, apps_by_type, as_dict=True, include_any_app=False):
-        used_case_types = set()
-        case_types_by_app = collections.defaultdict(list)
-        for app_type, apps in apps_by_type.items():
-            for app_choice in apps:
-                if app_choice.id not in [self.UNKNOWN_SOURCE, self.ALL_SOURCES]:
-                    app = get_app(self.domain, app_choice.id)
-                    case_types = []
-                    if hasattr(app, 'modules'):
-                        # Add regular case types
-                        case_types = set([
-                            module.case_type
-                            for module in app.modules if module.case_type
-                        ])
-
-                        # Add user case if any module uses it
-                        if any([module.uses_usercase() for module in app.modules]):
-                            case_types.add(USERCASE_TYPE)
-
-                        used_case_types = used_case_types.union(case_types)
-                        case_types = [RMIDataChoice(
-                            id=c,
-                            text=c,
-                            data=app_choice.data
-                        ) for c in case_types]
-                        if as_dict:
-                            case_types = [c._asdict() for c in case_types]
-                    case_types_by_app[app_choice.id] = case_types
-
-        all_case_types = get_case_types_for_domain_es(self.domain)
-        unknown_case_types = all_case_types.difference(used_case_types)
-        unknown_case_types = [RMIDataChoice(
-            id=c,
-            text=c,
+    def _get_cases_in_domain(self, as_dict=True):
+        all_case_type_names = get_case_types_for_domain(self.domain)
+        all_case_type_objs = [RMIDataChoice(
+            id=case_type,
+            text=case_type,
             data={
-                'unknown': True,
+                'unknown': True
             }
-        ) for c in unknown_case_types]
+        ) for case_type in all_case_type_names]
 
-        # Add all case types, along with "Select All" placeholder choice to
-        # the "Any Application" application placeholder
-        all_case_types = []
-        if include_any_app:
-            all_case_types = used_case_types
-            all_case_types = [RMIDataChoice(
-                id=c,
-                text=c,
+        all_case_type_objs.insert(
+            0,
+            RMIDataChoice(
+                id=ALL_CASE_TYPE_EXPORT,
```

Comment:
@esoergel Should 'All Case Types' be added here if only no. of case types > 1 or should always be there?

Reply:
I think it should always be there, though maybe you can help check my reasoning:

If "all case types" is selected, then the export config stores something representing "all".  This means that if future case types are created, they'll appear in the export automatically.  Conversely, if we instead stored the case types present at the time the export was created, then it shouldn't make a difference.

@zandre-eng does that sound right to you too?

Reply:
@esoergel That sounds correct. The `ALL_CASE_TYPE_EXPORT` is simply a defined tag that we use to identify bulk case exports. Since we only grab the list of actual case types when downloading the export, I agree that this should always be there as the number of case types could change at a later stage.

Reply:
Thanks @esoergel and @zandre-eng for the clarification.

Reply:
@esoergel Should the above mentioned change be done for 'All Case Types' or leave it as it is?

Reply:
@zandre-eng do you think that behavior should be changed, then?  I think it's fine to display "All Case Types" even if there's only one, though I don't feel strongly about it.

Reply:
@esoergel I think from a user's perspective it would make more sense to not show "All Case Types" if there is only one. Otherwise, users might wonder what they are bulk exporting since they only have one case type available.

Reply:
that's fine by me

Reply:
https://github.com/dimagi/commcare-hq/pull/33088/commits/3e3efec6f7423c71b388046b9adf701ac08fdf5d